frontend/src/reducers/credentials.js

```diff
@@ -0,0 +1,78 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { batch } from 'react-redux'
+import { API_URL } from '../reusables/urls'
+
+const credentials = createSlice({
+    name: 'credentials',
+    initialState: {
+        username: null,
+        accessToken: null,
+        error: null,
+        secret: ''
+    },
+    reducers: {
+        setUsername: (store, action) => {
+            store.username = action.payload
+        },
+        setAccessToken: (store, action) => {
+            store.accessToken = action.payload
+        },
+        setError: (store, action) => {
+            store.error = action.payload
+        },
+        logOut: (store, action) => {
+            store.username = null
+            store.accessToken = null
+        },
+        setSecret: (store, action) => {
+            console.log(action.payload)
+            store.secret = action.payload
+        }
+    }
+})
+
+export const authenticate = (username, password, mode) => {
+    return (dispatch, getState) => {
+        const state = getState()
+        //const method = mode === 'secret' ? 'GET' : 'POST'
+        const options = mode === 'secret' 
+        ? {
+            method: 'GET',
+            headers: {
+            'Content-type' : 'application/json',
+            'Authorization' : state.credentials.accessToken
+            }
+        }
+        : {
+            method: 'POST',
+            headers: {
+            'Content-type' : 'application/json',
+            'Authorization' : state.credentials.accessToken
+            },
+            body: JSON.stringify({ username, password })
+        }
```

Comment:
That is a really long ternary operator that is very difficult to read. Perhaps, consider if statements?